test/hotspot/gtest/runtime/test_committed_virtualmemory.cpp

```diff
@@ -214,3 +214,70 @@ TEST_VM(CommittedVirtualMemoryTracker, test_committed_virtualmemory_region) {
   }
 
 }
+
+#if !defined(_WINDOWS) && !defined(_AIX)
```

Comment:
I might be doing something wrong, but this test consistently fails on my windows machine. The full committed region is returned even if it is untouched. If I try committing a huge amount of memory (that is untouched), `os::committed_in_range `will return the full region, although actual RSS has not increased. So I think `VirtualQuery` does indeed return the amount committed, but does not distinguish between paged-in or not, which is what this test intends to check.

Reply:
As a side note, maybe `os::committed_in_range` should have a different name. Perhaps the name is a bit misleading because we actually intend for this method to return the subset of committed memory that is paged-in. It's also a bit confusing that it does the _intended_ behaviour on some platforms but does the _suggested_ behaviour  on other platforms (on windows it seems). Sometimes it does neither the intended or suggested behaviour and just returns the whole reserved region (AIX).

Reply:
> I might be doing something wrong, but this test consistently fails on my windows machine. The full committed region is returned even if it is untouched. If I try committing a huge amount of memory (that is untouched), `os::committed_in_range `will return the full region, although actual RSS has not increased. So I think `VirtualQuery` does indeed return the amount committed, but does not distinguish between paged-in or not, which is what this test intends to check.

We probably should query MEM_MAPPED, not MEM_COMMIT, after VirtualQuery 

> As a side note, maybe `os::committed_in_range` should have a different name. Perhaps the name is a bit misleading because we actually intend for this method to return the subset of committed memory that is paged-in. It's also a bit confusing that it does the _intended_ behaviour on some platforms but does the _suggested_ behaviour on other platforms (on windows it seems). Sometimes it does neither the intended or suggested behaviour and just returns the whole reserved region (AIX).

True. Lets deal with that in a follow-up, and disable the tests for windows for now.

I opened https://bugs.openjdk.org/browse/JDK-8334403 and put it on your name :)

Reply:
ok sounds good to me. Thanks!